File: lc/c/004.c
int maxArea(int* height, int heightSize){
    int ret = 0;
    int l = 0, r = heightSize-1;
    while (l < r) {
        int h = height[l] < height[r] ? height[l] : height[r];
        int w = r-l;
        int area = h * w;
        ret = area > ret ? area : ret;
        if (height[l] < height[r])
            ++l;
        else
            --r;
    }
    return ret;
}

File: lc/py/004.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ret = 0
        l, r = 0, len(height)-1
        area = lambda x,y: x*y
        while l < r:
            ret = max(ret, area(min(height[l], height[r]), r-l))
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return ret
